### ISSUE TITLE
Tabs keyboard operation

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7140]
### Description

Regarding the tabs UI which can be found at the top of CP pages where tabs are defined, e.g. the user settings pages and entry edit pages (when more than 1 tab of content has been configured).

<img width="820" alt="Screenshot of tabs along top of a user's account settings page. Tabs for: account, permissions, and preferences." src="https://user-images.githubusercontent.com/17884198/99225300-ae71b100-279c-11eb-8558-0bf09da525c1.png">

Visually, this UI appears like it should behave like tabs. That is, instead of using the <kbd>tab</kbd> key to progress through each tab item like a list of links, pressing <kbd>tab</kbd> should move focus to the active tab in the tablist and the left and right arrow keys should be used to switch between them.

Two resources from WAI-ARIA showing how to structure tabs:
- [Example of Tabs with Automatic Activation](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html)
- [Example of Tabs with Manual Activation](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html)

I think it is okay to leave the items as anchor elements, though, for the additional functionality of updating the hash in the URL.

On deciding between automatic and manual activation, [the WAI-ARIA best practices recommend to using automatic activation when there is no latency expected for changing the tab content](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_selection_follows_focus). I think that the tab content is all in the DOM on page load, so automatic activation should be fine.

Might also want to consider outlining the inactive tablist items so it's more visually clear that they are tabs rather than a list of links?

When there's many tabs, the ones that don't fit in the tabs list get put in a dropdown menu. However, when one of the items in the dropdown is selected, that selected item replaces the last visible tab. This reordering may be confusing. Maybe follow a pattern similar to this one: [Web accessibility tutorials: Fly-out Menus](https://www.w3.org/WAI/tutorials/menus/flyout/)

**Comment by missmatsuko** [comment]
Related: #7813

**Comment by brandonkelly** [comment]
> Visually, this UI appears like it should behave like tabs. That is, instead of using the <kbd>tab</kbd> key to progress through each tab item like a list of links, pressing <kbd>tab</kbd> should move focus to the active tab in the tablist and the left and right arrow keys should be used to switch between them.

Some pages have tabs that link to entirely different pages (e.g. Settings → Sections → Edit), and I don’t think anyone would expect that <kbd>←</kbd> / <kbd>→</kbd> keys would cause a navigation change. And we probably shouldn’t have different interaction models for the tabs depending on whether they link out to another page or not.

**Comment by missmatsuko** [comment]
Hmm I'm not sure if I'm looking at the right place, but we can go for manual activation if that makes more sense for our tabs.

Example here:
https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-2/tabs.html

Arrow keys are still used move the focus on the tabs, but enter is needed to activate it.

**Comment by brandonkelly** [comment]
Ahhh, interesting. That works for me.

**Comment by brandonkelly** [comment]
Done for the next release!
